### PR TITLE
Remove microbatch nan check

### PR DESCRIPTION
**Pull request by abhinavgoel95**
This PR creates an option to remove the NaN check at the end of every microbatch when using Pipeline Parallelism. This allows cublas to fuse the GeMMs with the gradient accumulation leading to a 6% training performance improvement on NVIDIA GPUs.

### CODE REVIEW

**Comment by ukoxyz** [comment]
This change is not safe. Layers that behave correctly could produce reasonable gradients on valid iterations, but NaN on bubble iterations.

**Comment by abhinavgoel95** [comment]
We set the default to false, we will only set it to true in the configs in https://github.com/google/paxml/blob/main/paxml/tasks/lm/params/nvidia.py

**Comment by nluehr** [comment]
@ukoxyz PTAL.

**Comment by zhangqiaorjc** [comment]
@nluehr mentioned that the nans in bubble iterations seen might be a TPU-specific behavior. Let's add this option so GPU users can enable it.
